frontend/libs/studio-components/src/components/StudioTextResourceInput/StudioTextResourceInput.tsx

```diff
@@ -1,156 +1,214 @@
-    </div>
-  );
-}
+StudioTextResourceInput.displayName = 'StudioTextResourceInput';
 
-enum InputMode {
+enum Mode {
   EditValue = 'editValue',
   Search = 'search',
 }
 
 type InputBoxProps = StudioTextResourceInputProps & {
-  inputMode: InputMode;
+  mode: Mode;
 };
 
-function InputBox({
-  currentId,
-  inputMode,
-  onChangeCurrentId,
-  onChangeTextResource,
-  textResources,
-  texts,
-}: InputBoxProps): ReactElement {
-  const currentTextResource = getTextResourceById(textResources, currentId);
-
-  switch (inputMode) {
-    case InputMode.EditValue:
-      return (
-        <ValueField
-          label={texts.valueLabel}
-          onChange={onChangeTextResource}
-          textResource={currentTextResource}
-        />
-      );
-    case InputMode.Search:
-      return (
-        <StudioTextResourcePicker
-          className={classes.inputbox}
-          emptyListText={texts.emptyResourceList}
-          label={texts.textResourcePickerLabel}
-          onValueChange={onChangeCurrentId}
-          textResources={textResources}
-          value={currentId}
-        />
-      );
-  }
-}
+const InputBox = forwardRef<HTMLInputElement, InputBoxProps>(
+  (
+    {
+      currentId,
+      inputClass,
+      mode,
+      onChangeCurrentId,
+      onChangeTextResource,
+      onKeyDown,
+      textResources,
+      texts,
+      ...rest
+    },
+    ref,
+  ): ReactElement => {
+    const currentTextResource = getTextResourceById(textResources, currentId);
+    const className = cn(inputClass, classes.inputbox);
+
+    switch (mode) {
+      case Mode.EditValue:
+        return (
+          <ValueField
+            className={className}
+            label={texts.valueLabel}
+            onChangeTextResource={onChangeTextResource}
+            onKeyDown={onKeyDown}
+            ref={ref}
+            textResource={currentTextResource}
+            {...rest}
+          />
+        );
+      case Mode.Search:
+        return (
+          <StudioTextResourcePicker
+            className={className}
+            emptyListText={texts.emptyResourceList}
+            label={texts.textResourcePickerLabel}
+            onValueChange={onChangeCurrentId}
+            onKeyDown={onKeyDown}
+            textResources={textResources}
+            ref={ref}
+            value={currentId}
+            {...rest}
+          />
+        );
+    }
+  },
+);
+
+InputBox.displayName = 'InputBox';
 
 type ValueFieldProps = {
-  label: string;
   textResource: TextResource;
-  onChange: (textResource: TextResource) => void;
-};
-
-function ValueField({ textResource, onChange, label }: ValueFieldProps): ReactElement {
-  const handleChange = (event: ChangeEvent<HTMLInputElement>) => {
-    const { value } = event.target;
-    const newTextResource = editTextResourceValue(textResource, value);
-    onChange(newTextResource);
-  };
+  onChangeTextResource: (textResource: TextResource) => void;
+} & StudioTextfieldProps;
```

Comment:
Kjempebagatell, men hadde det bitt cleanere om typen det arves fra står først her også som for `InputBoxProps` på linje 92?

Reply:
Ja, her gjør vi det litt forskjellig rundt omkring, men det er jo i hvert fall fint å holde det konsistent innad i filen. Fikser det!